generated/nidcpower/nidcpower/lcr_measurement.py

```diff
@@ -0,0 +1,229 @@
+    ]
+
+    def __init__(self, data):
+        """LCRMeasurement
+
+        Creates and returns an LCRMeasurement object.
+
+        Args:
+            data (struct_NILCRMeasurement): The LCR measurement ctypes object returned by the driver.
+        """
+        self.channel = ""
+        self.vdc = data.vdc
+        self.idc = data.idc
+        self.stimulus_frequency = data.stimulus_frequency
+        self.ac_voltage = complex(data.ac_voltage_real, data.ac_voltage_imaginary)
+        self.ac_current = complex(data.ac_current_real, data.ac_current_imaginary)
+        self.z = complex(data.z_real, data.z_imaginary)
+        self.series_lcr = LCRMeasurement.LCR(
+            inductance=data.ls, capacitance=data.cs, resistance=data.rs
+        )
+        self.parallel_lcr = LCRMeasurement.LCR(
+            inductance=data.lp, capacitance=data.cp, resistance=data.rp
+        )
+        self.d = data.d
+        self.measurement_mode = enums.InstrumentMode(data.measurement_mode)
+        self.in_compliances = LCRMeasurement.InCompliances(
+            dc=bool(data.dc_in_compliance), ac=bool(data.ac_in_compliance)
+        )
+        self.unbalanced = bool(data.unbalanced)
+
+    @property
+    def z_magnitude_and_phase(self):
+        """z_magnitude_and_phase
+
+        Returns a tuple of (z_magnitude, z_phase) of this LCRMeasurement object.
+
+        Returns:
+            z_magnitude_and_phase (tuple of (float, float)):
+
+                - **z_magnitude** (float): The magnitude of the complex impedance, in ohms.
+                - **z_phase** (float): The impedance phase angle, in degrees.
+
+        """
+        z_magnitude, z_phase_in_radians = cmath.polar(self.z)
+        return z_magnitude, math.degrees(z_phase_in_radians)
+
+    @property
+    def y(self):
+        """y
+
+        Returns the admittance of this LCRMeasurement object.
+
+        Returns:
+            y (complex): The complex admittance.
+
+        """
+        if self.z == 0:
+            return complex(cmath.nan, cmath.nan)
+        return 1.0 / self.z
+
+    @property
+    def y_magnitude_and_phase(self):
+        """y_magnitude_and_phase
+
+        Returns a tuple of (y_magnitude, y_phase) of this LCRMeasurement object.
+
+        Returns:
+            y_magnitude_and_phase (tuple of (float, float)):
+
+                - **y_magnitude** (float): The magnitude of the complex admittance, in siemens.
+                - **y_phase** (float): The admittance phase angle, in degrees.
+
+        """
+        y_magnitude, y_phase_in_radians = cmath.polar(self.y)
+        return y_magnitude, math.degrees(y_phase_in_radians)
+
+    @property
+    def q(self):
+        """q
+
+        Returns the quality factor of this LCRMeasurement object.
+
+        Returns:
+            q (float): The quality factor of the circuit.
+
+        """
+        if self.d == 0:
+            return math.nan
+        return 1.0 / self.d
```

Comment:
If you have self.d and that has q as computed by the driver... why compute here again?

Reply:
Are you mistaking `self.d` as `data`? `self.d` here is referring to dissipation factor, which is the reciprocal of `q`.

`q` is not stored as a member of `LCRMeasurement` as decided during our previous LCR Python API review meeting, so it has to be calculated / derived on-demand here.

Reply:
I did mistake it. But I do think it's a good idea to store _data and remove calculations done in this code.
As much as possible, this is a thin wrapper over the C API and the math should be kept to an absolute minimum. In this case, we can have the same desired API without doing the math because the info is already in `data` as calculated by the driver runtime.